### PR TITLE
Update GO dependencies for 2.53

### PR DESCRIPTION
**Pull request by krajorama**
Ran "make update-all-go-deps" , but remove GO version bump and  toolchain clause from go.mod

### CODE REVIEW

File: discovery/ovhcloud/ovhcloud_test.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ endpoint: %s
 
 	_, err := createClient(&conf)
 
-	require.ErrorContains(t, err, "missing application key")
```
Comment (krajorama) [inline discovery/ovhcloud/ovhcloud_test.go:69]: note: due to github.com/ovh/go-ovh update v1.4.3 to v1.5.1

**Comment by krajorama** [comment]
From 2.53 release: FYI @roidelapluie @remyleone @Heyoxe as people that worked on Scaleway SD.  I've deferred the update of the Scaleway sdk due to non trivial (to me) changes in it. See https://github.com/prometheus/prometheus/pull/14212/commits/777daea86e0ee2cee1e00f3cd75293bde9aa7f51

**Comment by krajorama** [comment]
Dependency update changelogs:
https://github.com/KimMachineGun/automemlimit/releases/tag/v0.6.1 (small log fix)

AWS SDK minor version bumps don't contain relevant removals:
https://github.com/aws/aws-sdk-go/releases/tag/v1.53.0
https://github.com/aws/aws-sdk-go/releases/tag/v1.52.0

godo: only additions
https://github.com/digitalocean/godo/releases/tag/v1.116.0
https://github.com/digitalocean/godo/releases/tag/v1.115.0
https://github.com/digitalocean/godo/releases/tag/v1.114.0

https://github.com/gophercloud/gophercloud/releases/tag/v1.12.0 fixes only

grafana/regexp : https://github.com/grafana/regexp/pull/10

github.com/hashicorp/consul/api no dedicated changelog, trusting semver
github.com/hashicorp/nomad/api only commit ids, trusting tests, no major version change in the meantime

https://github.com/linode/linodego/releases/tag/v1.34.0 has a breaking change, but I don't believe it impacts us (new acl field returned)
https://github.com/linode/linodego/releases/tag/v1.35.0 nothing to see here

https://github.com/ovh/go-ovh/releases/tag/v1.5.0 new feature oath2
https://github.com/ovh/go-ovh/releases/tag/v1.5.1 fix

Rest is more low level and indirect.

**Comment by remyleone** [comment]
@krajorama We are going to look into it. We are in a transition for instance between nat ip and routed ip. How open would you be to just silence the anti deprecated linter while we are still in the transition? @Codelax

**Comment by krajorama** [comment]
> @krajorama We are going to look into it. We are in a transition for instance between nat ip and routed ip. How open would you be to just silence the anti deprecated linter while we are still in the transition? @Codelax

I'm not really familiar with Scaleway, so if you're fine with it, I can add some `//nolint` comments - especially now that you're aware. The tricky part will actually be the change from a single public ip to multiple ips. I'll make a PR to bump the dependency and just silence linter by adding the comment.

**Review by bwplotka — APPROVED** [review]
Thanks, and thanks for due diligence on change log, LGTM!

**Comment by remyleone** [comment]
https://github.com/prometheus/prometheus/pull/14228

**Pull request merged**